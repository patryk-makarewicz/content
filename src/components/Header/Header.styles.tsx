import styled, { css } from 'styled-components';

import { colors } from '../../styles/colors';
import { breakpoint, zIndex } from '../../styles/variables';

type InnerWrapperProps = {
  $justifyEnd?: boolean;
};

type MenuProps = {
  close: boolean;
};

export const Config = styled.div`
  position: relative;
  max-width: 1440px;
  height: 35px;
  margin: auto;
  border-bottom: 1px solid ${colors.primary};
  background-color: var(--background-secondary-color);
  z-index: ${zIndex.header};
`;

export const InnerWrapper = styled.nav<InnerWrapperProps>`
  display: flex;
  align-items: center;
  ${({ $justifyEnd }) =>
    $justifyEnd
      ? css`
          justify-content: flex-end;
        `
      : css`
          justify-content: space-between;
        `}
  width: 100%;
  padding: 0 10px;
  margin: auto;
  height: 100%;

  @media screen and (min-width: ${breakpoint.s}) {
    ${({ $justifyEnd }) =>
      $justifyEnd
        ? css`
            justify-content: flex-end;
          `
        : css`
            justify-content: flex-start;
          `}
  }
`;

export const Header = styled.header`
  max-width: 1440px;
  height: 60px;
  margin: auto;
  border-bottom: 2px solid var(--border-color);
  background-color: var(--background-secondary-color);
  position: sticky;
  top: 0px;
  z-index: ${zIndex.header};
`;

export const MobileMenu = styled.div<MenuProps>`
  background-color: var(--background-secondary-color);
  color: var(--text-color);
  position: fixed;
  top: 35px;
  left: auto;
  min-width: 100%;
  min-height: 100%;
  transition: all 0.2s ease-out;
  z-index: ${zIndex.mobileMenu};
  overflow: hidden;
  padding-top: 100px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;

  ${({ close }) =>
    close
      ? css`
          right: -100%;
          visibility: hidden;
        `
      : css`
          right: 0;
          visibility: visible;
        `}
`;

export const FlagButton = styled.button`
  background-color: inherit;
  text-decoration: none;
  border: none;
  cursor: pointer;
  display: flex;
`;
