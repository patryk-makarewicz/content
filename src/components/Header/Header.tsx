import { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';

import { Hamburger } from 'components/Hamburger';
import { Navigation } from 'components/Navigation';
import { ThemeButton } from 'components/ThemeButton';

import { Logo } from 'utils/logo';
import { useWindowSize } from 'utils/windowSize';

import FlagEn from '../../assets/flag_en.png';
import { ReactComponent as FlagPl } from '../../assets/flag_poland.svg';
import * as Styled from './Header.styles';

export const Header = () => {
  const { i18n } = useTranslation();

  const [showMenu, setShowMenu] = useState(false);
  const { width } = useWindowSize();

  const onOpenCloseMenu = () => {
    setShowMenu((prevState) => !prevState);
  };

  const [language, setLanguage] = useState('en');
  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
    setLanguage(lng);
  };

  const onChangeLanguage = () => {
    if (language === 'pl') {
      changeLanguage('en');
    } else {
      changeLanguage('pl');
    }
  };

  const renderFlag = () => {
    if (language === 'en') {
      return <img src={FlagEn} alt="Language flag En" width={24} height={22} style={{ marginLeft: '3px' }} />;
    } else {
      return <FlagPl width={27} height={32} />;
    }
  };

  useEffect(() => {
    setShowMenu(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [width >= 576]);

  return (
    <>
      <Styled.Config id="about">
        <Styled.InnerWrapper $justifyEnd>
          <ThemeButton />
          <Styled.FlagButton onClick={onChangeLanguage} style={{ marginLeft: '15px', padding: 0 }}>
            {renderFlag()}
          </Styled.FlagButton>
        </Styled.InnerWrapper>
      </Styled.Config>

      <Styled.Header>
        <Styled.InnerWrapper>
          <h1 style={{ margin: 'auto 15px auto 0', display: 'flex' }}>{Logo()}</h1>
          <Navigation />
          <Hamburger showMenu={showMenu} onOpenCloseMenu={onOpenCloseMenu} />
        </Styled.InnerWrapper>
      </Styled.Header>

      <Styled.MobileMenu close={!showMenu}>
        <Navigation onClick={() => setShowMenu(false)} mobile />
      </Styled.MobileMenu>
    </>
  );
};
