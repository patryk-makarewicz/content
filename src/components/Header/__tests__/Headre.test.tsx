import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { RecoilRoot } from 'recoil';

import { Header } from '../Header';

const renderHeader = () => {
  const { baseElement, getAllByText } = render(
    <Router>
      <RecoilRoot>
        <Header />
      </RecoilRoot>
    </Router>
  );

  return {
    baseElement,
    getAllByText,
  };
};

describe('test Header', () => {
  it('should take the snapshot', () => {
    const { baseElement } = renderHeader();

    expect(baseElement).toMatchSnapshot();
  });

  it('should render navigation', () => {
    const { getAllByText } = renderHeader();

    // take from desktop navigation
    const navHome = getAllByText('navigation.home')[0];
    expect(navHome).toBeInTheDocument();
  });
});
