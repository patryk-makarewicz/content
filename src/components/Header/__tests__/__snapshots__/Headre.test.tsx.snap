// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header should take the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-jXbUNg eWkAXN"
      id="about"
    >
      <nav
        class="sc-dhKdcB kyuFJC"
      >
        <button
          class="sc-kAyceB vlONZ"
        >
          <div
            class="sc-imWYAI jxbXiK"
          />
          <img
            height="13"
            src="[object Object]"
            width="13"
          />
          <img
            height="15"
            src="[object Object]"
            width="15"
          />
        </button>
        <button
          class="sc-jlZhew fzAHDC"
          style="margin-left: 15px; padding: 0px;"
        >
          <img
            alt="Language flag En"
            height="22"
            src="test-file-stub"
            style="margin-left: 3px;"
            width="24"
          />
        </button>
      </nav>
    </div>
    <header
      class="sc-kpDqfm loVOVT"
    >
      <nav
        class="sc-dhKdcB fRyLRL"
      >
        <h1
          style="margin: auto 15px auto 0px; display: flex;"
        >
          <img
            height="40"
            src="[object Object]"
            width="35"
          />
        </h1>
        <nav>
          <ul
            class="sc-dcJsrY gmuWzF"
          >
            <li
              class="sc-iGgWBj btkQbK"
              data-testid="nav-navigation.home"
            >
              <a
                aria-current="page"
                class="sc-gsFSXq gCFTJR active"
                href="/"
              >
                navigation.home
              </a>
            </li>
            <li
              class="sc-iGgWBj btkQbK"
              data-testid="nav-navigation.content"
            >
              <a
                class="sc-gsFSXq gCFTJR"
                href="/content"
              >
                navigation.content
              </a>
            </li>
            <li
              class="sc-iGgWBj btkQbK"
              data-testid="nav-navigation.contact"
            >
              <a
                class="sc-gsFSXq gCFTJR"
                href="/contact"
              >
                navigation.contact
              </a>
            </li>
          </ul>
        </nav>
        <button
          class="sc-aXZVg Gvapt"
          data-testid="button-hamburger"
          type="button"
        >
          <div
            class="sc-gEvEer cfdbZP"
          />
          <div
            class="sc-gEvEer sc-eqUAAy cfdbZP giEXpz"
          />
          <div
            class="sc-gEvEer sc-fqkvVR cfdbZP eJjnVN"
          />
        </button>
      </nav>
    </header>
    <div
      class="sc-dAlyuH hNVXph"
    >
      <nav>
        <ul
          class="sc-dcJsrY gJZbaZ"
        >
          <li
            class="sc-iGgWBj dwxSLc"
            data-testid="nav-navigation.home"
          >
            <a
              aria-current="page"
              class="sc-gsFSXq gCFTJR active"
              href="/"
            >
              navigation.home
            </a>
          </li>
          <li
            class="sc-iGgWBj dwxSLc"
            data-testid="nav-navigation.content"
          >
            <a
              class="sc-gsFSXq gCFTJR"
              href="/content"
            >
              navigation.content
            </a>
          </li>
          <li
            class="sc-iGgWBj dwxSLc"
            data-testid="nav-navigation.contact"
          >
            <a
              class="sc-gsFSXq gCFTJR"
              href="/contact"
            >
              navigation.contact
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</body>
`;
